packages/venia-ui/lib/components/FilterModal/filterBlock.js

```diff
@@ -1,118 +1,55 @@
-            ? classes.optionNameExpanded
-            : classes.optionName;
-
-        return (
-            <div className={classes.optionHeader}>
-                <button
-                    onClick={this.optionToggle}
-                    className={classes.optionToggleButton}
-                >
-                    <span className={nameClass}>{item.name}</span>
-                    <span className={classes.closeWrapper}>
-                        <Icon src={iconSrc} />
-                    </span>
-                </button>
-            </div>
-        );
-    };
-
-    getLayout = options => {
-        const { layout } = options ? options : {};
-        const { classes } = this.props;
-        switch (layout) {
-            case filterLayouts.grid:
-                return classes.layoutGrid;
-            default:
-                return classes.layout;
-        }
-    };
-
-    getRenderOptions = value =>
-        filterRenderOptions[`${value}`] ||
-        filterRenderOptions[filterModes.default];
-
-    render() {
-        const {
-            classes,
-            item: { filter_items, request_var, name }
-        } = this.props;
+import { Form } from 'informed';
+import { useFilterBlock } from '@magento/peregrine/lib/talons/FilterModal';
+import setValidator from '@magento/peregrine/lib/validators/set';
 
-        const { isExpanded } = this.state;
-
-        const { mode, options } = this.getRenderOptions(request_var);
-
-        const listClassName = isExpanded
-            ? classes.filterListExpanded
-            : classes.filterList;
-
-        const controlBlock = this.getControlBlock(isExpanded);
-
-        const filterLayoutClass = this.getLayout(options);
-
-        const isSwatch = filterModes[mode] === filterModes.swatch;
-
-        const filterProps = {
-            isSwatch,
-            options,
-            name,
-            mode,
-            id: request_var,
-            items: filter_items,
-            layoutClass: filterLayoutClass
-        };
-
-        return (
-            <li className={classes.root}>
-                {controlBlock}
-                <div className={listClassName}>
-                    <FilterList {...filterProps} />
-                </div>
-            </li>
-        );
-    }
-}
+import { mergeClasses } from '../../classify';
+import Icon from '../Icon';
+import FilterList from './FilterList';
+import defaultClasses from './filterBlock.css';
 
-export default classify(defaultClasses)(FilterBlock);
+const FilterBlock = props => {
+    const { filterApi, filterState, group, items, name } = props;
+    const talonProps = useFilterBlock({ group });
+    const { handleClick, isExpanded, isSwatch } = talonProps;
+    const iconSrc = isExpanded ? ArrowUp : ArrowDown;
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const listProps = { filterApi, filterState, group, isSwatch, items, name };
+    const listElement = isExpanded ? <FilterList {...listProps} /> : null;
```

Comment:
Previously we were using CSS to hide the list. I did a little performance check and it takes 0.2ms to render the list on my mac.

It's _just_ enough that I noticed it. Not sure if you want to address that in the PR but it's something of note.

Reply:
Fixed it anyway. 👍